### DISCUSSION TITLE
把API-KEY放在源代码中而不是设置中是否可行？

### DISCUSSION THREAD

**Discussion by ghost** [ddiu8081/chatgpt-demo#381] [General]
把API-KEY放在源代码中而不是设置中是否可行？

**Comment by erbanku** [comment]
当然可行，但极其不安全，尤其是放公网。建议还是老老实实放环境变量里。
如果你把包含 `API Key` 的代码放公开仓库，还会导致你的 `API Key` 被作废（ OpenAI 官方几乎实时检测公开仓库 `API Key`），需要去重新生成。

**Reply by ghost** [reply]
Beca我想将其部署到我的域名上，无需在打开它时再输入API_KEY，即可立即获得响应。
